TodoApp.ConsoleApp/Repositories/XMLRepository/RepositoryBase.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+using TodoApp.ConsoleApp.Repositories.Interfaces;
+using TodoApp.Library.Data;
+using System.Xml;
+using System.Linq;
+using System.IO;
+using System.Data;
+
+namespace TodoApp.ConsoleApp.Repositories.Models.XmlRepository
+{
+    abstract class RepositoryBase<TModel, TId> : IRepository<TModel, TId> where TModel : Imodel<TId>
+    {
+        protected string Path { get; }
+        protected XDocument Document { get; }
+        protected XElement ContainerElement { get; }
+
+        protected RepositoryBase(string path,XName containerName)
+        {
+            Path = path;
+            Document = XDocument.Load(path);
+            ContainerElement = Document.Element(containerName);
+        }
+
+        protected abstract TModel ElementToEntity(XElement element);
+        protected abstract XElement EntityToElement(TModel entity);
+
+        public IEnumerable<TModel> GetAll()
+        {
+            List<TModel> models = new List<TModel>();
+
+            foreach (var element in XMLContent())
+            {
+                models.Add(ElementToEntity(element));
+            }
+            return models;
+        }
+
+        public void Delete(TId id)
+        {
+            ContainerElement.Elements().FirstOrDefault(a => a.Attribute("id").Value == id.ToString()).Remove();
```

Comment:
Split this into multiple lines.
Check if `FirstOrDefault` returns `null`.
The string `"id"` should be made abstract property of the class, so different models can have different name for the `"id"` field.

```suggestion
            ContainerElement.Elements()
                .FirstOrDefault(a => a.Attribute(IdName).Value == id.ToString())?
                .Remove();
```

EDIT: You should use the new method `GetElementById`